src/instr/instr_paje_trace.cpp

```diff
@@ -720,7 +729,26 @@ void PopStateEvent::print() {
     stream << " " << type->id << " " << container->id;
     print_row();
   } else if (instr_fmt_type == instr_fmt_TI) {
-    /* Nothing to do */
+    char *process_id;
+
+    instr_extra_data extra;
+    xbt_dynar_pop(state_tracker, &extra);
+    if(extra->type  == TRACING_CUSTOM){
+      if (strstr(container->name, "rank-") == nullptr)
+	process_id = xbt_strdup(container->name);
+      else
+	process_id = xbt_strdup(container->name + 5);
+
+      FILE *trace_file = (FILE *)xbt_dict_get(tracing_files, container->name);
+      if(extra->print_pop){
+        extra->print_pop(trace_file, process_id, extra);
```

Comment:
We should not have to care about this anymore; all we need is a custom subclass of TIData that is used instead of the "extra" structure; in this case, for instance in PMPI_Iteration_in() etc. (not implemented here, implemented in the LB code of rktesser).